File: android/app/src/main/java/com/aitutor/app/domain/model/Group.kt
package com.aitutor.app.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Room(
    val id: String,
    val name: String,
    @SerialName("created_by") val createdBy: String,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class GroupMessage(
    val id: String,
    @SerialName("room_id") val roomId: String,
    @SerialName("user_id") val userId: String,
    val content: String,
    @SerialName("display_name") val displayName: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

File: ios/AITutor/Domain/Model/Group.swift
import Foundation

struct Room: Codable, Equatable, Identifiable {
    let id: String
    let name: String
    let createdBy: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, name
        case createdBy = "created_by"
        case createdAt = "created_at"
    }
}

struct GroupMessage: Codable, Equatable, Identifiable {
    let id: String
    let roomId: String
    let userId: String
    let content: String
    let displayName: String?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, content
        case roomId = "room_id"
        case userId = "user_id"
        case displayName = "display_name"
        case createdAt = "created_at"
    }
}
